### PR TITLE
fix: revert state

### LINKED ISSUE #109: Not reverting storage states
**Describe the bug**
Failed to revert storage states.

_[Reported by Piotr at Halborn]_

**To Reproduce**
```
contract C {
  uint256 public num;
  function set1(uint256 x) public {
    num = x;
    revert("blah");
  }
}

contract CTest {
    function checkRevert1(uint256 x) public {
        C c = new C();
        require(x != 0);
        (bool result, ) = address(c).call(abi.encodeWithSignature("set1(uint256)", x));
        assert(!result);
        assert(c.num() != x); // fail
    }
}
```

### PR DESCRIPTION
**Pull request by daejunpark**
fix #109

### CODE REVIEW

File: src/halmos/sevm.py [hunk 1]
```diff
@@ -1261,6 +1261,11 @@ def call(
 
         caller = ex.prank.lookup(ex.this, to)
 
+        orig_code = ex.code.copy()
+        orig_storage = deepcopy(ex.storage)
+        orig_balance = deepcopy(ex.balance)
+        orig_log = deepcopy(ex.log)
```
Comment (karmacoma-eth) [inline src/halmos/sevm.py:1267]: note to self: I have an idea to get rid of all these deepcopies, I'll get on it soon

File: tests/test/Revert.t.sol [hunk 2]
```diff
@@ -37,4 +43,21 @@ contract CTest {
         assert(!result);
         assert(c.num() != x);
     }
+
+    function check_RevertBalance(bool paused, uint256 amount) public {
```
Comment (karmacoma-eth) [inline tests/test/Revert.t.sol:47]: cool! do we also have a check to verify that `code` is properly reverted too?

Comment (karmacoma-eth) [reply tests/test/Revert.t.sol:47]: ah I see https://github.com/a16z/halmos/issues/145 👀

**Review by karmacoma-eth — APPROVED** [review]

File: tests/test/Revert.t.sol (hunk 2 above)
Comment (daejunpark) [reply tests/test/Revert.t.sol:54]: #145 is a different one.  i added a test for reverting code info.

**Review by karmacoma-eth — APPROVED** [review]
👌

**Pull request merged**